### ISSUE TITLE
DeviceOption::set_device_type() doesn't accept caffe2::CPU anymore

### ISSUE THREAD

**Issue by SPark9625** [pytorch/pytorch#14408]
### TL;DR
It'd be best if someone could write up an up-do-date method on loading a (pretrained) caffe2 network in C++. I've searched for the past couple of days, but all methods that I've tried don't work with the current pytorch version ('1.0.0a0+60e7d04', from the master branch)

But at the same time I understand that pytorch v1.0 is still a preview and there could be big API changes, in which case I would like to ask a small question. How, in its current state, am I supposed to call the `DeviceOption::set_device_type()` function? It seems like it used to be by passing in `c10::DeviceType`, but now it's expecting an integer (`::google::protobuf::int32`). But I saw no documentation on this.

---

### My situation

I have a model that I trained in Pytorch, in Python. Using onnx, I converted my model, and saved it as two different .pb files using the `caffe2.python.backend.Caffe2Backend.onnx_graph_to_caffe2_net()` function, and then saving it `with open(...)`. Now I'm trying to load it in C++ caffe2. Here's my current code.

```
int main() {

    caffe2::NetDef init_net, pred_net;

    std::string init_path = "10B_2018-11-27_10-27-38_init.pb";
    std::string pred_path = "10B_2018-11-27_10-27-38_pred.pb";


    CAFFE_ENFORCE(caffe2::ReadProtoFromFile(init_path, &init_net));
    CAFFE_ENFORCE(caffe2::ReadProtoFromFile(pred_path, &pred_net));

    init_net.mutable_device_option()->set_device_type(<???>);
                                                       ^^^
    pred_net.mutable_device_option()->set_device_type(<???>);
                                                       ^^^
```

Now when I look at some previous sources, like [this](https://github.com/caffe2/caffe2/issues/1737) or [this](https://github.com/BIGBALLON/Caffe2_Demo/blob/master/03_cpp_forward/main.cpp), I'm supposed to pass in `caffe2::CPU`, which is of type `const c10::DeviceType`, but currently in Pytorch '1.0.0a0+60e7d04', this function expects type `'::google::protobuf::int32' (aka 'int')`, but **there is no documentation** on which integer represents CPU or CUDA etc..

Furthermore, some (more recent) resources suggest saving it directly as an onnx model and loading it with caffe2 backend (c.f., [TRANSFERING A MODEL FROM PYTORCH TO CAFFE2 AND MOBILE USING ONNX](https://pytorch.org/tutorials/advanced/super_resolution_with_caffe2.html)), I haven't seen anything similar in C++. Is it not yet supported?

Any help would be highly appreciated. Please help!

**Comment by ezyang** [comment]
In this particular case, the enum values of `DeviceType` (e.g., `DeviceType::CPU`) are in one-to-one correspondence with the expected protobuf values. The documentation situation is less than ideal, so I'll keep this issue open for that.

**Comment by SPark9625** [comment]
Aha, I tried to put in `caffe2::CPU`, but was keep getting compilation errors, and the reason was because `caffe2::DeviceType` is an enum **class**. After reading @ezyang ‘s answer, I made an explicit conversion to `int` and things are working now. :)
